.buildbot.sh

```diff
@@ -0,0 +1,10 @@
+#/bin/sh
+
+set -e
+
+export CHERI=$HOME/cheri/output/sdk
```

Comment:
Is this saying "install into $CHERI" or does it mean "pick the build up from $CHERI"?

Reply:
the latter

Reply:
OK, this is probably not quite doing what we think it's doing then :) Basically, I *think* what's happened is that you've got *n* buildbot users (if it's anything like b13, *n*=4) and you're kind of getting lucky if you've installed a `cheri` directory into that buildbot user's home directory. It's not a reliable way of doing things unfortunately.

There are at least two "better" ways of doing things:

1. Install cheri from scratch in each buildbot session. This is the easiest, but obviously slowest, thing to do.
2. Cache a cheri build somewhere (on b13 we tend to pick somewhere in `/opt`) and copy it over in each buildbot session.

Reply:
Oh my configuration isn't anything like b13, I use docker which does things differently.
The way it actually works is very interesting, every night a new docker container is build.
It runs cherybuild.py and produces a filesystem snapshot that holds the sdk in ~/cheri/outpu/.
Then I can spin up an infinite (figuratively) number of worker instances and all of them will have the 
cheri directory in the right place because all of them use the same underlying read-only filesystem snapshot. 
And after the test the temporary mutations get discarded.

Reply:
Ah, OK, that's the second option in disguise then :)

Just to be clear: *every* time buildbot finishes, the docker image is destroyed?

Reply:
Yes a very fancy version of the second option, without copying. 
Not the image, the container. The image stays until a new one is build.
The container is temporary execution of the image lets say, which also
holds the mutable data. 
btw this is how I have it setup, it is possible to use the same container multiple 
time but it is as cheap to just create a new one, and discard the old one, and also 
that causes less problems.

Reply:
Just to be clear: if I do 100 buildbot builds in a single day, those builds won't be able to persist any data that a later build can see?

Reply:
Yes